foreman/data_refinery_foreman/surveyor/harmony.py

```diff
@@ -585,25 +761,42 @@ def harmonize_value(self, field_name: str, value):
         else:
             return value.lower().strip()
 
-    def harmonize_field(self, sample_metadata: Dict, harmonized_sample: Dict, field: str):
-        field_name = field.split("_fields")[0]
+    def harmonize_field(
+        self, sample_metadata: Dict, harmonized_sample: Dict, variants_attribute: str
+    ):
+        sample_attribute = variants_attribute.split("_fields")[0]
+        harmonized_values = set()
+        key_variants = getattr(self, variants_attribute)
+        key_regexes = [r for r in key_variants if isinstance(r, RE_TYPE)]
+        key_strings = set([s for s in key_variants if isinstance(s, str)])
```

Comment:
```suggestion
        key_strings = set((s for s in key_variants if isinstance(s, str)))
```

or 

`key_strings = {s for s in key_variants if isinstance(s, str)}`